app/imaging/svg.go

```
@@ -20,39 +20,68 @@ type SVGInfo struct {
 
 // ParseSVG returns information for the given SVG input data.
 func ParseSVG(svgReader io.Reader) (SVGInfo, error) {
-	var parsedSVG struct {
-		Width   string `xml:"width,attr,omitempty"`
-		Height  string `xml:"height,attr,omitempty"`
-		ViewBox string `xml:"viewBox,attr,omitempty"`
-	}
 	svgInfo := SVGInfo{
 		Width:  0,
 		Height: 0,
 	}
-	viewBoxPattern := regexp.MustCompile("^([0-9]+)[, ]+([0-9]+)[, ]+([0-9]+)[, ]+([0-9]+)$")
-	dimensionPattern := regexp.MustCompile("(?i)^([0-9]+)(?:px)?$")
 
-	// decode provided SVG
-	if err := xml.NewDecoder(svgReader).Decode(&parsedSVG); err != nil {
-		return svgInfo, err
-	}
+	decoder := xml.NewDecoder(svgReader)
 
-	// prefer viewbox for SVG dimensions over width/height
-	if viewBoxMatches := viewBoxPattern.FindStringSubmatch(parsedSVG.ViewBox); len(viewBoxMatches) == 5 {
-		svgInfo.Width, _ = strconv.Atoi(viewBoxMatches[3])
-		svgInfo.Height, _ = strconv.Atoi(viewBoxMatches[4])
-	} else if parsedSVG.Width != "" && parsedSVG.Height != "" {
-		widthMatches := dimensionPattern.FindStringSubmatch(parsedSVG.Width)
-		heightMatches := dimensionPattern.FindStringSubmatch(parsedSVG.Height)
-		if len(widthMatches) == 2 && len(heightMatches) == 2 {
-			svgInfo.Width, _ = strconv.Atoi(widthMatches[1])
-			svgInfo.Height, _ = strconv.Atoi(heightMatches[1])
+	for {
+		token, err := decoder.Token()
+		if err != nil {
+			return svgInfo, err
 		}
-	}
+		switch t := token.(type) {
+		case xml.StartElement:
+			for _, attr := range t.Attr {
+				if attr.Name.Local == "viewBox" {
+					values := strings.Fields(attr.Value)
+					if len(values) == 4 {
+						width := 0
+						_, err := fmt.Sscan(values[2], &width)
+
+						if err != nil {
+							svgInfo.Width = width
```

Are you meaning to not ignore the errors in the `"viewBox"` case?